src/views/AmalgamationSelection.vue

```diff
@@ -171,63 +184,77 @@ export default class AmalgamationSelection extends Vue {
     return 'Unknown'
   }
 
-  /** Start Regular Long-form button pressed. */
+  /** Called when Start Regular Long-form button is clicked. */
   async startRegularAmalgamation (): Promise<any> {
-    const legalType = this.getLegalType as CorpTypeCd
-
-    // Create a draft amalgamation and redirect to Create UI
+    // Create a draft amalgamation application then redirect to Create UI.
     try {
       // show spinner since this is a network call
       this.setStartingAmalgamationSpinner(true)
-      const accountId = +JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT'))?.id || 0
-      const businessId = await this.createBusinessAA(accountId, legalType, AmalgamationTypes.REGULAR)
+      const businessId = await this.createBusinessAA(AmalgamationTypes.REGULAR)
       const amalgamationUrl = `${this.getCreateUrl}?id=${businessId}`
       navigate(amalgamationUrl)
       return
     } catch (error) {
+      console.log('Error: unable to amalgamate now =', error)
       this.setStartingAmalgamationSpinner(false)
-      throw new Error('Unable to Amalgamate Now ' + error)
+      this.showErrorDialog = true
     }
   }
 
   /**
-   * Create a draft amalgamation application based on selected business type.
-   * @param accountId Account ID of logged in user.
-   * @param legalType The legal type of the amalgamated business
+   * Creates a draft amalgamation application.
+   * @param type the type of amalgamation to create
+   * @returns the business identifier of the newly created amalgamation application
    */
-  async createBusinessAA (accountId: number, legalType: CorpTypeCd, type: AmalgamationTypes): Promise<string> {
-    const businessRequest = {
+  private async createBusinessAA (type: AmalgamationTypes): Promise<string> {
+    const accountId = +JSON.parse(sessionStorage.getItem('CURRENT_ACCOUNT'))?.id || 0
+    const legalType = this.getLegalType
+
+    const draftAmalgamationApplication = {
       filing: {
         header: {
           name: FilingTypes.AMALGAMATION_APPLICATION,
-          accountId: accountId
+          accountId
         },
         business: {
-          legalType: legalType
+          legalType
         },
         amalgamationApplication: {
           nameRequest: {
-            legalType: legalType
+            legalType
           },
-          type: type
+          type
         }
       }
     } as any
 
-    const createBusinessResponse =
-      await LegalServices.createBusiness(businessRequest).catch(error => {
-        throw new Error('Unable to create new Amalgamation Draft ' + error)
-      })
+    // if this is a regular amalgamation, pre-populate the current business as a TING
+    if (type === AmalgamationTypes.REGULAR) {
+      draftAmalgamationApplication.filing.amalgamationApplication.amalgamatingBusinesses = [
+        {
+          type: AmlTypes.LEAR,
+          role: AmlRoles.AMALGAMATING,
+          identifier: this.getIdentifier
+        }
+      ]
+    }
```

Comment:
We should be able to extend this for short-form amalgamations to add the holding/primary company.